tests/any_item/any_item.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "gtest/gtest.h"
+
+#include <libcellml>
+
+#include "test_utils.h"
+
+TEST(AnyItemPtr, createDefault)
+{
+    auto anyItem = libcellml::AnyItem::create();
+    EXPECT_EQ(libcellml::CellmlElementType::UNDEFINED, anyItem->type());
+}
+
+TEST(AnyItemPtr, createWithArguments)
+{
+    auto model = libcellml::Model::create();
+    auto item = libcellml::AnyItem::create(libcellml::CellmlElementType::MODEL, std::any(model));
+    EXPECT_EQ(libcellml::CellmlElementType::MODEL, item->type());
+    EXPECT_EQ(model, std::any_cast<libcellml::ModelPtr>(item->item()));
+}
```

Comment:
Needs test for all API of the class `setItem` and `setType`.

Reply:
I have removed `setItem()` and `setType()` from the `AnyCellmlElement` API. Indeed, I reckon that a user should only ever consume an `AnyCellmlElement` object, not modify its contents. The latter should only be made possible internally (by the `Annotator` class).

Reply:
Not relevant anymore.